File: MovieServer/MovieServer.API/MovieServer.Domain/src/Repositories/ICartRepo.ts
import { Cart } from "@Domain/Entities";
import { CartId } from "@Domain/ValueObjects";

export default interface ICartRepo {
    Get(id: CartId): Promise<Cart | null>;
    Create(cart: Cart): Promise<any>;
    Update(cart: Cart): Promise<any>;
    Remove(cart: Cart): Promise<any>;
}

File: OrderServer/OrderServer.API/src/OrderServer.Domain/Repositories/ICartRepo.cs
using OrderServer.Domain.Entities;
using OrderServer.Domain.ValueObjects;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OrderServer.Domain.Repositories
{
    public interface ICartRepo
    {
        Task<Cart?> GetAsync(CartId id);
        Task DeleteAsync(Cart cart);
    }
}
